### PR TITLE
PUBDEV-8654: Added documentation on monotone splines for GAM

### PR DESCRIPTION
**Pull request by hannah-tillman**
For: [PUBDEV-8654](https://h2oai.atlassian.net/browse/PUBDEV-8654)

This PR adds documentation on the monotone spline for GAM. I've attached a pdf of the local build for all the content that got added. Please let me know if anything else needs to be updated/changed!

[Monotone-splines.pdf](https://github.com/h2oai/h2o-3/files/8475103/Monotone-splines.pdf)

### CODE REVIEW

File: h2o-docs/src/product/data-science/gam.rst [hunk 1]
```diff
@@ -401,6 +401,114 @@ For documentation on thin plate regression splines, refer here:
...
+
+.. math::
+   
+   penaltyMat = \int_{t_0}^{t_N}I_k^{2ndDeriv} \times \text{ transpose of }(I_k^{2ndDeriv})dt
+
+Element at row :math:`m` and column :math:`n` of :math:`penaltyMat` is
+
+.. math::
+   
+   penaltyMat_{m,n} = \int_{t_0}^{t_N}{\frac{d^2(I_{m,k}(t))}{d^2t}}{\frac{d^2(I_{n,k}(t))}{d^2t}}dt
```
Comment (wendycwong) [inline h2o-docs/src/product/data-science/gam.rst:511]: Hannah:  Not sure where you can add this but please let the user know that they have to specify monotone spline by specifying bs=2 and then specify the spline_orders parameter.

File: h2o-docs/src/product/data-science/gam.rst [hunk 2]
```diff
@@ -579,3 +689,15 @@ References
...
+
+3. Lecture 7 Divided Difference Interpolation Polynomial by Professor R.Usha, Department of Mathematics, IITM, https://www.youtube.com/watch?v=4m5AKnseSyI .
+
+.. _ref4:
+
+4. Carl De Boor et. al., ON CALCULATING WITH B-SPLINES II. INTEGRATION, ResearchGate Article, January 1976.
+
+.. _ref5:
+
+5. J.O. Ramsay, “Monotone Regression Splines in Action”, Statistical Science, 1988, Vol. 3, No. 4, 425-461.
```
Comment (wendycwong) [inline h2o-docs/src/product/data-science/gam.rst:703]: Can you add a python/R example on how to use monotone spline like before for the other spline types?

**Review by wendycwong — CHANGES_REQUESTED** [review]
Please also add the changes for GLMV3.java

File: h2o-docs/src/product/data-science/gam.rst (hunk 1 above)
Comment (hannah-tillman) [reply h2o-docs/src/product/data-science/gam.rst:511]: Added in the opening

File: h2o-docs/src/product/data-science/gam.rst [hunk 3]
```diff
@@ -401,6 +401,114 @@ For documentation on thin plate regression splines, refer here:
...
 
+Monotone Splines
+''''''''''''''''
+
+We have implemented I-splines which are used as monotone splines. Monotone splines do not support multinomial or ordinal families. In order to specify the monotone spline, you need to both set ``bs=2`` and specify the ``spline_orders`` parameter.
+
+**B-splines:** :math:`Q_{i,k}(t)`
+
+This implementation begins with the B-spline. Let :math:`Q_{i,k}(t)` denote a B-spline of order :math:`k` at knot :math:`i` where it is non-zero over the duration :math:`t_i \leq t < t_{i+k}`. The recursive formula [:ref:`4<ref4>`] used to generate a B-spline of a higher order from a B-spline of a lower order is:
+
```
Comment (wendycwong) [inline h2o-docs/src/product/data-science/gam.rst:412]: change

from a B-spline

to 

from two B-splnes

**Review by wendycwong — APPROVED** [review]

**Pull request merged**